streamz/sources.py

```diff
@@ -467,6 +468,30 @@ def __init__(self, topic, consumer_params, poll_interval='1s',
     def poll_kafka(self):
         import confluent_kafka as ck
 
+        def commit(_part):
+            topic, part_no, _, _, offset = _part[1:]
+            _tp = ck.TopicPartition(topic, part_no, offset + 1)
+            self.consumer.commit(offsets=[_tp], asynchronous=True)
+
+        @gen.coroutine
+        def checkpoint_emit(_part):
+            ref = RefCounter(cb=lambda: commit(_part))
+            yield self._emit(_part, metadata={'refs': ref})
+
+        tps = []
+        for partition in range(self.npartitions):
+            tps.append(ck.TopicPartition(self.topic, partition))
+
+        while True:
+            try:
+                committed = self.consumer.committed(tps, timeout=1)
+            except ck.KafkaException:
+                pass
```

Comment:
Does an exception here cause rapid-fire attempts to commit? This might seize the whole program

Reply:
**This** is the line that is causing coverage to go down...

Reply:
This could cause a rapid fire requests of already committed offsets. I suppose an alternative is to either try once or a specified number of times. If this fails n times, then we could honor the `auto.offset.reset` configuration in the `consumer_params` arugment. 

What do you think?